### DISCUSSION TITLE
local setup problems [Jun 23]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#2520] [Discord Threads]
> 🙋‍♂️ **moi**

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### lets discuss here  
#  
> 🙋‍♂️ **wkw**  
#### > weird, with node 16 I could not install and it seems I had to be on node v8 to have npm install working almost properly   
####    
#### can you show how you installed n what kind of errors you had  
#  
> 🙋‍♂️ **wkw**  
#### > It seems I also need python2. I think the documentation you gave me is used for something else   
####    
#### why? are you looking at development setup?   
#### https://docs.nocodb.com/getting-started/installation/#development-setup  
#  
> 🙋‍♂️ **moi**  
#### I succeed to install the backend with node 16  
#  
> 🙋‍♂️ **moi**  
#### I discovered the issue passing with node8 and having some python errors  
#  
> 🙋‍♂️ **moi**  
#### So I had to install pyenv et python2.7 so I could install with node 16  
#  
> 🙋‍♂️ **moi**  
#### I had some node-gyp errors  
#  
> 🙋‍♂️ **wkw**  
#### why node 8 ..  
#  
> 🙋‍♂️ **moi**  
#### an error I get suggest to use node 8 so I tried  
#  
> 🙋‍♂️ **moi**  
#### maybe I missread the errors but it helped me fix it 🙂  
#  
> 🙋‍♂️ **wkw**  
#### do u mean npm version 8  
#  
> 🙋‍♂️ **wkw**  
#### my local setup    
####    
#### node v16.14.0   
#### npm v8.3.1  
#  
> 🙋‍♂️ **wkw**  
#### u may try follow this  
#  
> 🙋‍♂️ **moi**  
#### No, I change my node version, maybe the errors handler using the python is not compatible with node16  
#  
> 🙋‍♂️ **moi**  
#### I did `n 8` (n is an alternative to nvm)  
#  
> 🙋‍♂️ **wkw**  
#### node 8 failed for sure  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/989082657893789696)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
